src/physics_mg.cpp

```diff
@@ -0,0 +1,240 @@
+  if (material_xs.absorption > 0.) {
+    absorption(p, material_xs);
+  } else {
+    p->absorb_wgt = 0.;
+  }
+  if (!p->alive) return;
+
+  // Sample a scattering event to determine the energy of the exiting neutron
+  scatter(p, energy_bin_avg);
+
+  // Play Russian roulette if survival biasing is turned on
+  if (settings::survival_biasing) {
+    russian_roulette(p);
+    if (!p->alive) return;
+  }
+}
+
+void
+scatter(Particle* p, const double* energy_bin_avg)
+{
+  // Adjust indices for Fortran to C++ indexing
+  // TODO: Remove when no longer needed
+  int gin = p->last_g - 1;
+  int gout = p->g - 1;
+  int i_mat = p->material - 1;
+  macro_xs[i_mat].sample_scatter(gin, gout, p->mu, p->wgt);
+
+  // Adjust return value for fortran indexing
+  // TODO: Remove when no longer needed
+  p->g = gout + 1;
+
+  // Rotate the angle
+  rotate_angle_c(p->coord[0].uvw, p->mu, nullptr);
+
+  // Update energy value for downstream compatability (in tallying)
+  p->E = energy_bin_avg[gout];
+
+  // Set event component
+  p->event = EVENT_SCATTER;
+}
+
+void
+create_fission_sites(Particle* p, Bank* bank_array, int64_t& size_bank,
+                     const int64_t& bank_array_size,
+                     const MaterialMacroXS& material_xs)
+{
+  // TODO: Heat generation from fission
+
+  // If uniform fission source weighting is turned on, we increase or decrease
+  // the expected number of fission sites produced
+  double weight;
+  if (settings::ufs_on) {
+    weight = ufs_get_weight(p);
+  } else {
+    weight = 1.;
+  }
+
+  // Determine the expected number of neutrons produced
+  double nu_t = p->wgt / openmc_keff * weight * material_xs.nu_fission /
+       material_xs.total;
+
+  // Sample the number of neutrons produced
+  int nu = static_cast<int>(nu_t);
+  if (prn() <= (nu_t - int(nu_t))) {
+    nu++;
+  }
+
+  // Check for the bank size getting hit. For fixed source calculations, this
+  // is a fatal error; for eigenvalue calculations, it just means that k-eff
+  // was too high for a single batch.
+  if (size_bank + nu > bank_array_size) {
+    if (settings::run_mode == RUN_MODE_FIXEDSOURCE) {
+      throw std::runtime_error{"Secondary particle bank size limit reached."
+           " If you are running a subcritical multiplication problem,"
+           " k-effective may be too close to one."};
+    } else {
+      if (mpi::master) {
+        std::stringstream msg;
+        msg << "Maximum number of sites in fission bank reached. This can"
+             " result in irreproducible results using different numbers of"
+             " processes/threads.";
+        warning(msg);
+      }
+    }
+  }
+
+  // Begin banking the source neutrons
+  // First, if our bank is full then don't continue
+  if ((nu == 0) || (size_bank == bank_array_size)) return;
+
+  // Initialize the counter of delayed neutrons encountered for each delayed
+  // group.
+  double nu_d[MAX_DELAYED_GROUPS] = {0.};
+
+  p->fission = true;
+  // TODO: +1 for start?
```

Comment:
What's this about?

Reply:
Ah a note I left for myself that i no longer need.  Whoops.